### ISSUE TITLE
CloudStack 4.19.0 can't export templates/ISOs by CloudStack UI

### ISSUE THREAD

**Issue by Dragonzz27** [apache/cloudstack#9463]
All i want is to download templdates already made from instances.

CloudStack 4.19.0 is deployed and all the components work fine.

~~~
ISOs are uploaded successfully and Templates are converted from volumes successfully.
~~~

![sunlogin202407291613100-imageonline co-4484359](https://github.com/user-attachments/assets/ad3d9e46-4f6e-4746-a432-9d51974e1725)

![sunlogin202407291613420-imageonline co-4189596](https://github.com/user-attachments/assets/779d42cc-9d31-45ed-85c8-51446710a2fc)

~~~
System vms are checked that they are running without anything unusual.
~~~

![sunlogin202407291614530-imageonline co-2166102](https://github.com/user-attachments/assets/fe672883-fba7-4c01-9125-62aea6bfd48b)

~~~
I try to export/download templates from CloudStack UI by the button 'download template' but failed.
wget/curl has been tried.
edge/chrome has been tried.
system vms has been restarted.
~~~

![sunlogin202407291614010-imageonline co-3547006](https://github.com/user-attachments/assets/2f8aa8c9-3c5e-46e5-b0ea-7028cb197832)

![sunlogin202407291614320-imageonline co-6015936](https://github.com/user-attachments/assets/dd0258fb-9c8e-4ff7-8c02-54f1422580cf)

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
CloudStack UI, API
~~~

##### CLOUDSTACK VERSION

~~~
CloudStack 4.19.0
~~~

##### CONFIGURATION

~~~
basic network
~~~

##### OS / ENVIRONMENT

~~~
CentOS 7.9.2009
~~~

##### SUMMARY

Download templates from CloudStack failed.

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
templates are downloaded successfully.
~~~

##### ACTUAL RESULTS

~~~
failed to download templates with timeout.
~~~

**Comment by kiranchavala** [comment]
@Dragonzz27 , I am not able to reproduce the behaviour

Could you please try with the extract template API call 

https://cloudstack.apache.org/api/apidocs-4.19/apis/extractTemplate.html

Example

```
(CMK) > extract template id=b451ccef-0ff5-4702-bcd1-66e73fe48450 zoneid=7625c6bd-67ee-4847-a3b4-48ee73281aed  mode=HTTP_DOWNLOAD
{
  "template": {
    "accountid": "80c25e48-2941-11ef-b10c-1e007c000297",
    "extractMode": "HTTP_DOWNLOAD",
    "id": "b451ccef-0ff5-4702-bcd1-66e73fe48450",
    "name": "kube-v1.27.2/ubuntu-2004",
    "state": "DOWNLOAD_URL_CREATED",
    "url": "http://10.0.54.122/userdata/7ca93d49-d6e0-4cfa-a755-a0b5133bf93b.qcow2",
    "zoneid": "7625c6bd-67ee-4847-a3b4-48ee73281aed",
    "zonename": "ref-trl-6823-k-Mr8-kiran-chavala"
  }
}
```

Then try wget the file 
                                                                                                                                                          
```
╰─ wget "http://10.0.54.122/userdata/7ca93d49-d6e0-4cfa-a755-a0b5133bf93b.qcow2"
--2024-07-29 15:12:04--  http://10.0.54.122/userdata/7ca93d49-d6e0-4cfa-a755-a0b5133bf93b.qcow2
Connecting to 10.0.54.122:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4399300608 (4.1G)
Saving to: ‘7ca93d49-d6e0-4cfa-a755-a0b5133bf93b.qcow2’

7ca93d49-d6e0-4cfa-a755-a0b5133bf93b.qcow2       0%[                                                                                                  ]   1.15M  1.04MB/s


```

**Comment by rajujith** [comment]
@Dragonzz27 Can you check the following: 

- Connectivity to the IP and port 80 of the URL. 
- Try from another client, network. 
- Check if the template is present in the secondary storage.

**Comment by Dragonzz27** [comment]
Really thanks to @kiranchavala and @rajujith , the problem has been solved and templates/ISOs can be downloaded successfully.

I checked two system vms following rajujith's suggestions and found that the consoleproxy vm could be pinged but secondarystorage vm could not be pinged successfully.

Although apache2 server in the vm looks working fine, but when curl http://IP_Address:80, it failed.

So I finally delete the secondarystorage vm and reinstall using the vmtemplate, with a result, I can download templates and ISOs using edge without any problem.

![sunlogin202407311236180-imageonline co-5432100](https://github.com/user-attachments/assets/8a9a13ce-7c43-4f90-ba52-d3db1180fce0)
